pymeasure/instruments/ametek/ametek7270.py

```diff
@@ -50,161 +71,242 @@ class Ametek7270(Instrument):
         value on DAC4 in Volts. This property can be set. """,
         validator=truncated_range,
-        values=[-10, 10]
+        values=[-10, 10],
+        check_set_errors=True,
     )
     adc1 = Instrument.measurement("ADC. 1",
-                                  """ Reads the input value of ADC1 in Volts """
+                                  """ Reads the input value of ADC1 in Volts """,
+                                  get_process=check_read_not_empty,
                                   )
     adc2 = Instrument.measurement("ADC. 2",
-                                  """ Reads the input value of ADC2 in Volts """
+                                  """ Reads the input value of ADC2 in Volts """,
+                                  get_process=check_read_not_empty,
                                   )
     adc3 = Instrument.measurement("ADC. 3",
-                                  """ Reads the input value of ADC3 in Volts """
+                                  """ Reads the input value of ADC3 in Volts """,
+                                  get_process=check_read_not_empty,
                                   )
     adc4 = Instrument.measurement("ADC. 4",
-                                  """ Reads the input value of ADC4 in Volts """
+                                  """ Reads the input value of ADC4 in Volts """,
+                                  get_process=check_read_not_empty,
                                   )
     id = Instrument.measurement("ID",
-                                """ Reads the instrument identification """
+                                """ Reads the instrument identification """,
+                                cast=str,
                                 )
 
-    def __init__(self, adapter, name="Ametek DSP 7270", **kwargs):
+    def __init__(self, adapter, name="Ametek DSP 7270",
+                 read_termination='\x00',
+                 write_termination='\x00',
+                 **kwargs,
+                 ):
+
         super().__init__(
             adapter,
             name,
-            **kwargs
+            read_termination=read_termination,
+            write_termination=write_termination,
+            **kwargs,
         )
 
+    def check_errors(self):
+        """mandatory to be used for property setter
+
+        The Ametek protocol expect the default null character to be read to check the property
+        has been correctly set. With default termination character set as Null character,
+        this turns out as an empty string to be read.
+        """
+        if self.read() == '':
+            return []
+        else:
+            return ['Incorrect return from previously set property']
+
+    def ask(self, command, query_delay=0):
+        """Usually the properties use the `values` method that adds a strip call,
+        however several methods use directly the result from ask to be casted into some other types.
+        It should therefore also add the strip here
+        """

```

Comment:
```suggestion
        """Send a command and read the response, stripping white spaces.

        Usually the properties use the `values` method that adds a strip call,
        however several methods use directly the result from ask to be casted into some other types.
        It should therefore also add the strip here, as all responses end with a newline character.
        """
```
I reformatted the docstring according to https://peps.python.org/pep-0257/

First a summary line, then a more extensive description (here I included, why it is necessary).